app-modules/form/src/Models/FormRequest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2022-2023, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace Assist\Form\Models;
+
+use App\Models\User;
+use App\Models\BaseModel;
+use Illuminate\Database\Eloquent\Builder;
+use Assist\Form\Enums\FormRequestDeliveryMethod;
+use Illuminate\Database\Eloquent\Relations\MorphTo;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+/**
+ * @mixin IdeHelperFormRequest
+ */
+class FormRequest extends BaseModel
+{
+    protected $fillable = [
+        'canceled_at',
+        'form_id',
+        'method',
+        'note',
+    ];
+
+    protected $casts = [
+        'canceled_at' => 'timestamp',
+        'method' => FormRequestDeliveryMethod::class,
+    ];
+
+    public function form(): BelongsTo
+    {
+        return $this->belongsTo(Form::class, 'form_id');
+    }
+
+    public function recipient(): MorphTo
+    {
+        return $this->morphTo();
+    }
+
+    public function submission(): BelongsTo
+    {
+        return $this->belongsTo(FormSubmission::class, 'submission_id');
+    }
+
+    public function user(): BelongsTo
+    {
+        return $this->belongsTo(User::class);
+    }
+
+    public function deliver(): void
+    {
+        $this->method->deliver($this);
+    }
+
+    public function scopeNotCanceled(Builder $query): Builder
+    {
+        return $query->whereNull('canceled_at');
+    }
```

Comment:
We may want to create something like an ENUM for `FormSubmissionStatus` that has `open`, `canceled`, and `submitted` as options with a method here or in the ENUM that does the logic determination of what the current status is.

Else, we could end up doing that check-in multiple places in the app.

Reply:
To be honest, I disagree. Currently, the status is calculable based on the `canceled_at`, and the `submission_id`. Adding in an extra column would represent that same information again, requiring it to be kept up to date. Also, the checks for these (two `whereNull()` column checks) are so simple that I don't even think it would be inconvenient?

Reply:
I don't think we should add a column. Simply a method that returns the status of the `FormRequest` as an ENUM based on the logic you described. So that if we need to check the status of a `FormRequest` there is a central place that is determined.

Reply:
Done